src/content/docs/node/introduction.mdx

```diff
@@ -1,3 +1,502 @@
+const fs = require("fs");
+
+fs.readFile("file.txt", "utf8", (err, data) => {
+  if (err) {
+    console.error("Error reading file", err);
+    return;
+  }
+  console.log(data);
+});
+```
+
+Here, the **readFile** function reads the file asynchronously, and the callback is executed when the file is read. If there’s an error, it's handled inside the callback.
+
+## 2. Promises
+
+A **promise** is an object that represents the eventual completion (or failure) of an asynchronous operation and its resulting value. It provides a cleaner way to work with async code compared to callbacks.
+
+```js
+const readFilePromise = fs.promises.readFile("file.txt", "utf8");
+
+readFilePromise
+  .then((data) => {
+    console.log(data);
+  })
+  .catch((err) => {
+    console.error("Error reading file", err);
+  });
+```
+
+Here, **readFilePromise** handles the asynchronous file read, and we use **.then()** to handle the success and **.catch()** to handle errors. Promises allow you to chain multiple async operations in a readable way.
+
+## 2. Async/Await
+
+**Async/await** is a more modern and cleaner way to write asynchronous code in JavaScript. It is built on promises but allows you to write asynchronous code that looks synchronous.
+
+```js
+const fs = require("fs").promises;
+
+async function readFileAsync() {
+  try {
+    const data = await fs.readFile("file.txt", "utf8");
+    console.log(data);
+  } catch (err) {
+    console.error("Error reading file", err);
+  }
+}
+
+readFileAsync();
+```
+
+In this example, **await** pauses the execution of the function until the promise resolves, making the code easier to read. If an error occurs, it’s handled with **try/catch**.
+
+# Modules and the Module System in Node.js
+
+In Node.js, the module system allows developers to structure and reuse code efficiently. A module is essentially a file containing code (functions, objects, or variables) that can be exported and imported into other files. Node.js follows a modular architecture, enabling developers to divide code into smaller, maintainable parts.
+
+#### Creating and Exporting a Module
+
+To create a module, you define the logic you want to share in a JavaScript file. For example, let's create a module that exports a simple function:
+
+```js
+//math.js
+
+const fs = require("fs").promises;
+function add(a, b) {
+  return a + b;
+}
+
+// Export the function
+module.exports = { add };
+```
+
+Here, the add function is exported using module.exports, which makes it available to other files.
+
+**Importing a Module**
+Once a module is created and exported, it can be imported and used in other files using the require function:
+
+```js
+//app.js
+
+const math = require("./math");
+console.log(math.add(2, 3)); // Output: 5
+```
+
+### CommonJS vs. ES Modules
+
+Node.js supports two module systems: **CommonJS** (the older, default module system) and **ES Modules** (a more modern, standards-based approach). Let's compare their differences:
+
+#### Syntax Comparison
```

Comment:
This is not required since this is given in the below table. Therefore instead of Syntax Comparison give some **heading for table** and remove this

```
#### Syntax Comparison
CommonJS uses `module.exports` and `require` for exporting and importing modules, while ES Modules use `export` and `import`.
```